components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -142,11 +142,14 @@ private ClaimUtil() {
 
             boolean isSubjectClaimInRequested = false;
             if (subjectClaimURI != null || requestedLocalClaimMap != null && requestedLocalClaimMap.length > 0) {
```

Comment:
would it make sense to change this condition to,
`if (subjectClaimURI != null && requestedLocalClaimMap != null && requestedLocalClaimMap.length > 0)`

Then we can avoid the repeated null check for requestedLocalClaimMap

Reply:
Following login should be executed when `subjectClaimURI != null` even the requestedLocalClaimMap is null. So the combined '&' cannot be used